Example/PNDecryptionService/NotificationService.swift

```diff
@@ -62,57 +62,62 @@ class NotificationService: UNNotificationServiceExtension {
-            outgoingMessageType: .outgoingMessageText,
-            content: pushMessage.body,
-            speakableGroupName: nil,
-            conversationIdentifier: pushMessage.type,
-            serviceName: nil,
-            sender: senderPerson,
-            attachments: []
-        )
-
-        incomingMessagingIntent.setImage(senderAvatar, forParameterNamed: \.sender)
-
-        let interaction = INInteraction(intent: incomingMessagingIntent, response: nil)
-        interaction.direction = .incoming
-        interaction.donate(completion: nil)
-
-        return try content.updating(from: incomingMessagingIntent)
+    func handle(content: UNNotificationContent, pushMessage: NotifyMessage, topic: String) -> UNNotificationContent {
+        do {
+            let iconUrl = try pushMessage.icon.asURL()
+
+            let senderThumbnailImageData = try Data(contentsOf: iconUrl)
+            let senderThumbnailImageFileUrl = try downloadAttachment(data: senderThumbnailImageData, fileName: iconUrl.lastPathComponent)
+            let senderThumbnailImageFileData = try Data(contentsOf: senderThumbnailImageFileUrl)
+            let senderAvatar = INImage(imageData: senderThumbnailImageFileData)
+
+            var personNameComponents = PersonNameComponents()
+            personNameComponents.nickname = pushMessage.title
+
+            let senderPerson = INPerson(
+                personHandle: INPersonHandle(value: topic, type: .unknown),
+                nameComponents: personNameComponents,
+                displayName: pushMessage.title,
+                image: senderAvatar,
+                contactIdentifier: nil,
+                customIdentifier: topic,
+                isMe: false,
+                suggestionType: .none
+            )
+
+            let selfPerson = INPerson(
+                personHandle: INPersonHandle(value: "0", type: .unknown),
+                nameComponents: nil,
+                displayName: nil,
+                image: nil,
+                contactIdentifier: nil,
+                customIdentifier: nil,
+                isMe: true,
+                suggestionType: .none
+            )
+
+            let incomingMessagingIntent = INSendMessageIntent(
+                recipients: [selfPerson],
+                outgoingMessageType: .outgoingMessageText,
+                content: pushMessage.body,
+                speakableGroupName: nil,
+                conversationIdentifier: pushMessage.type,
+                serviceName: nil,
+                sender: senderPerson,
+                attachments: []
+            )
+
+            incomingMessagingIntent.setImage(senderAvatar, forParameterNamed: \.sender)
+
+            let interaction = INInteraction(intent: incomingMessagingIntent, response: nil)
+            interaction.direction = .incoming
+            interaction.donate(completion: nil)
+
+            return try content.updating(from: incomingMessagingIntent)
+        } 
+        catch {
```

Comment:
This feels like a hack to wrap the whole thing in a try/catch. Can't you check for `""` and then disable the logic?

Reply:
Execution also can failed on: 
1. downloading resource (network problem)
2. accessing device storage

Before that fix all of that cases failing message decoding

Reply:
Nice that we found that